src/Sulu/Bundle/AdminBundle/Resources/js/components/Modal/ClickModal.js

```diff
@@ -0,0 +1,40 @@
+// @flow
+import React from 'react';
+import type {Element} from 'react';
+import {action, observable} from 'mobx';
+import {observer} from 'mobx-react';
+import type {ModalProps} from './types';
+import Modal from './Modal';
+
+type Props = ModalProps & {
+    className?: string,
+    clickElement: Element<*>,
+};
+
+@observer
+export default class ClickModal extends React.PureComponent<Props> {
```

Comment:
I am still not a big fan of this component... Don't you think it would make more sense to implement this as a high order component?